### ISSUE TITLE
Total Cost of an Empire

### ISSUE THREAD

**Issue by lsandig** [ehsan66/ICAOD#1]
Hi,
while trying to understand the details of how ICAOD works, I noticed a discrepancy between the paper and the implementation. Eq. (13) in the [article](https://dx.doi.org/10.1016/j.csda.2016.06.014) gives the total cost of an empire as
```math
Q_i = c_i - \zeta \frac{1}{N_i} \sum_{r=1}^{N_i} d_{r,i}
```
with
- $Q_i$ the total cost of empire $i$,
- $c_i$ the cost of imperialist $i$,
- $d_{r,i}$ the cost of colony $r$ of empire $i$,
- $N_i$ the number of colonies in empire $i$,
- and a default $\zeta=0.1$.

On the other hand, the implementation seems to be essentially
```r
ImperialistCost + zeta * mean(ColoniesCost)
```
https://github.com/ehsan66/ICAOD/blob/08a95cf543c6ac7dbd216efe5f85f02783d87538/R/2-InternalCommonFunctions.R#L131
https://github.com/ehsan66/ICAOD/blob/08a95cf543c6ac7dbd216efe5f85f02783d87538/R/2-InternalCommonFunctions.R#L385
https://github.com/ehsan66/ICAOD/blob/08a95cf543c6ac7dbd216efe5f85f02783d87538/R/2-InternalCommonFunctions.R#L535
https://github.com/ehsan66/ICAOD/blob/08a95cf543c6ac7dbd216efe5f85f02783d87538/R/3-UserMinimaxFunctions.R#L2705
https://github.com/ehsan66/ICAOD/blob/08a95cf543c6ac7dbd216efe5f85f02783d87538/R/6-UserBayesFunctions.R#L1103

Have I overlooked a flipped sign somewhere, or is this an error?

**Comment by ehsan66** [comment]
Both versions should be correct. As I remember, in the manuscript for the sake of simplicity for a minmax problem, the algorithm for a maximization problem was intruduced. Here, it could be the minimazion problem.
But I dont have the manuscript to check.

**Comment by lsandig** [comment]
I don't think it is due to the difference between a minimization / maximization problem. When we flip the sign of the objective function, this should affect both of $c_i$ and $d_{r,i}$, and not just one of them.
FWIW, [Atashpaz-Gargari/Lucas (2007)](https://ieeexplore.ieee.org/document/4425083)  and also [Lin et al. (2013)](https://www.scientific.net/AMM.284-287.3135.pdf), which you cite, have the "`+`" variant.
